### PR TITLE
tests: mcuboot: add boot_request_upgrade() return value check

### PR DESCRIPTION
**Pull request by butok**
- Adds boot_request_upgrade() return value check.
- Avoid repeating resets if the upgrade request fails.

### CODE REVIEW

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by hakehuang** [comment]
@butok I meet failure on

lpcexpresso55s36
```
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-4091-g5b04c6606fde ***
I: Starting bootloader
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-4091-g5b04c6606fde ***
I: Starting bootloader
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-4091-g5b04c6606fde ***
I: Starting bootloader
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-4091-g5b04c6606fde ***
I: Starting bootloader
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-4091-g5b04c6606fde ***
I: Starting bootloader

```

File: tests/boot/test_mcuboot/src/main.c [hunk 1]
```diff
@@ -11,10 +11,16 @@
...
 	/* Perform a permanent swap of MCUBoot application */
-	boot_request_upgrade(1);
-	printk("Secondary application ready for swap, rebooting\n");
-	sys_reboot(SYS_REBOOT_COLD);
+	err = boot_request_upgrade(BOOT_UPGRADE_PERMANENT);
+	if (err) {
+		printk("Failed to request upgrade: %d", err);
+	} else {
+		printk("Secondary application ready for swap, rebooting\n");
+		sys_reboot(SYS_REBOOT_COLD);
```
Comment (nordicjm) [inline tests/boot/test_mcuboot/src/main.c:23]: Might be worth adding a k_sleep() here for 200ms or so to ensure message is printed (optional)

**Review by nordicjm — APPROVED** [review]

File: tests/boot/test_mcuboot/src/main.c (hunk 1 above)
Comment (butok) [reply tests/boot/test_mcuboot/src/main.c:23]: > Might be worth adding a k_sleep() here for 200ms or so to ensure message is printed (optional)

It 's not needed. It is printed on all tested platforms.

**Comment by hakehuang** [comment]
ok, maybe my board issue atagin.

**Review by hakehuang — APPROVED** [review]

**Comment by butok** [comment]
Hi @d3zd3z , PR is waiting for you.

**Review by de-nordic — APPROVED** [review]

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**